spec/controllers/people_controller_spec.rb

```diff
@@ -20,139 +20,30 @@
+        expect(assigns(:person).send(field)).to eq(person.send(field))
       end
     end
 
-    context "with invalid params" do
-      it "assigns a newly created but unsaved person as @person" do
-        post :create, params: {person: invalid_attributes}, session: valid_session
-        expect(assigns(:person)).to be_a_new(Person)
-      end
-
-      it "re-renders the 'new' template" do
-        post :create, params: {person: invalid_attributes}, session: valid_session
-        expect(response).to render_template("new")
-      end
-    end
-  end
-
-  describe "PUT #update" do
-    context "with valid params" do
-      let(:new_attributes) {
-        skip("Add a hash of attributes valid for your model")
-      }
-
-      it "updates the requested person" do
-        person = Person.create! valid_attributes
-        put :update, params: {id: person.to_param, person: new_attributes}, session: valid_session
-        person.reload
-        skip("Add assertions for updated state")
-      end
-
-      it "assigns the requested person as @person" do
-        person = Person.create! valid_attributes
-        put :update, params: {id: person.to_param, person: valid_attributes}, session: valid_session
-        expect(assigns(:person)).to eq(person)
-      end
-
-      it "redirects to the person" do
-        person = Person.create! valid_attributes
-        put :update, params: {id: person.to_param, person: valid_attributes}, session: valid_session
-        expect(response).to redirect_to(person)
-      end
-    end
-
-    context "with invalid params" do
-      it "assigns the person as @person" do
-        person = Person.create! valid_attributes
-        put :update, params: {id: person.to_param, person: invalid_attributes}, session: valid_session
-        expect(assigns(:person)).to eq(person)
-      end
-
-      it "re-renders the 'edit' template" do
-        person = Person.create! valid_attributes
-        put :update, params: {id: person.to_param, person: invalid_attributes}, session: valid_session
-        expect(response).to render_template("edit")
-      end
-    end
-  end
-
-  describe "DELETE #destroy" do
-    it "destroys the requested person" do
-      person = Person.create! valid_attributes
-      expect {
-        delete :destroy, params: {id: person.to_param}, session: valid_session
-      }.to change(Person, :count).by(-1)
+    it "renders the template show.json.jbuilder" do
+      expect(show_person).to be_successful
+      expect(show_person).to render_template("people/show")
     end
 
-    it "redirects to the people list" do
-      person = Person.create! valid_attributes
-      delete :destroy, params: {id: person.to_param}, session: valid_session
-      expect(response).to redirect_to(people_url)
+    it "returns an error code when an invalid Id is provided" do
+      get :show, params: { id: 0, format: :json }
+      expect(response).to have_http_status(404)
     end
```

Comment:
This is a small test and it works the way it is now, but it is a good candidate for using `context`. With Rspec contexts, you would do something like:

```
context "when an invalid id is provided" do
  subject(:show_person) { get :show, params: { id: 0, format: :json } }

  it "returns an error code" do
    show_person
    expect(response).to have_http_status(404)
  end
end
```